src/util/url.ts

```diff
@@ -13,6 +15,28 @@ function getBaseUrl(url: URL): string {
   return baseUrl;
 }
 
+function getEndpointPath(
+  endpoint: ConfigurablePaths,
+  realmPath?: string,
+  customPaths?: CustomPaths,
+): string {
+  const realmUrlPath = getRealmUrlPath(realmPath);
+  const defaultPaths = {
+    authenticate: `json/${realmUrlPath}/authenticate`,
+    authorize: `oauth2/${realmUrlPath}/authorize`,
+    accessToken: `oauth2/${realmUrlPath}/access_token`,
+    endSession: `oauth2/${realmUrlPath}/connect/endSession`,
+    userInfo: `oauth2/${realmUrlPath}/userinfo`,
+    revoke: `oauth2/${realmUrlPath}/token/revoke`,
+    sessions: `json/${realmUrlPath}/sessions/`,
+  };
+  if (customPaths && customPaths[endpoint]) {
+    return customPaths[endpoint];
+  } else {
+    return defaultPaths[endpoint];
+  }
+}
+
```

Comment:
This is the core of the customizable server path feature. All code paths that result in building a URL for fetch goes through this utility function.